Runtime/Interaction/SnapZone.cs

```diff
@@ -151,9 +149,46 @@ private void OnDrawGizmos()
 
         protected virtual void Update()
         {
-            if (socketActive && selectTarget == null)
+            if (showInteractableHoverMeshes)
+            {
+                showInteractableHoverMeshes = false;
+            }
+
+            if (ShowHighlightObject == false || shownHighlightObject == false && highlightInstance)
+            {
+                Destroy(highlightInstance);
+                highlightInstance = null;
+            }
+
+            if (highlightInstance == false && shownHighlightObject)
+            {
+                highlightInstance = Instantiate(shownHighlightObject, attachTransform, true);
+
+                highlightInstance.transform.localPosition = Vector3.zero;
+                Vector3 localScale = transform.localScale;
+                highlightInstance.transform.localScale = Vector3.Scale(shownHighlightObject.transform.localScale, new Vector3(1f / localScale.x, 1f / localScale.y, 1f/ localScale.z));
+                
+                Transform centerOfMass = highlightInstance.transform.Find("Center Of Mass");
+
+                highlightInstance.transform.rotation = transform.rotation;
+
+                foreach (Renderer highlightRenderer in highlightInstance.GetComponentsInChildren<Renderer>())
+                {
+                    for (int i = 0; i < highlightRenderer.sharedMaterials.Length; i++)
+                    {
+                        highlightRenderer.sharedMaterials[i] = previewMaterial;
+                    }
+                }
+            }
```

Comment:
Maybe would be good to instantiate the highlight since the snap zone is generated so it can be visible in the editor too.